python/cudf/cudf/api/types.py

```diff
@@ -27,6 +27,55 @@
 )
 
 
+def dtype(arbitrary):
+    """
+    Return the cuDF-supported dtype corresponding to `arbitrary`.
+
+    Inputs
+    ------
+    arbitrary: dtype or scalar-like
+
+    Returns
+    -------
+    dtype: the cuDF-supported dtype that best matches `arbitrary`
+    """
+    # first, try interpreting arbitrary as a NumPy dtype that we support:
+    try:
+        np_dtype = np.dtype(arbitrary)
+        if np_dtype.name == "float16":
+            np_dtype = np.dtype("float32")
+        elif np_dtype.kind in ("OU"):
+            np_dtype = np.dtype("object")
+    except TypeError:
+        pass
+    else:
+        if np_dtype.kind not in "biufUOMm":
```

Comment:
```suggestion
        if np_dtype not in cudf._lib.types.np_to_cudf_types:
```

To make this maintainable should we just lookup our `np`<->`libcudf` type-map here? This was any new dtype support added will automatically be supported here by `cudf.dtype`.

Reply:
Agree -- but it would be nicer if the source of truth was in a more obiously named constant. For exmaple, something like: `cudf._lib.types.SUPPORTED_NUMPY_TYPES`.

Reply:
+1 to have a `cudf._lib.types.SUPPORTED_NUMPY_TYPES`

Reply:
There's a slight problem here where `<M8` is an acceptable return type here, but it's not a `SUPPORTED_NUMPY_TYPE` (supported types are `<M8[unit]`).